src/ui/components/results/verticalresultscomponent.js

```diff
@@ -252,7 +249,7 @@ export default class VerticalResultsComponent extends Component {
     const hasAppliedFilters = this.appliedFilterNodes.length || this.nlpFilterNodes.length;
     const showResultsHeader = this.resultsHeaderOpts.showResultCount ||
       (this.resultsHeaderOpts.showAppliedFilters && hasAppliedFilters);
-    this.query = this.core.globalStorage.getState(StorageKeys.QUERY);
+    this.query = this.core.globalStorage.getState(StorageKeys.QUERY) || '';
```

Comment:
Is this necessary for this to work?

Reply:
no we can remove this line, I was just worried because the default return value of globalStorage.getState() is null. I think right now it's really hard if not impossible to get this to happen, but if this.query does somehow become null we end up appending `?query=null` to the verticalURL.

Reply:
In the spirit of keeping changes atomic, could we add this as a separate change? I think we want to keep this strictly as QA changes for v1.4.0

I'm not sure about the downstream of affects of handling a `query=null` as a `query=` instead

Reply:
sure, will take it out